docs/layer_util.js

```diff
@@ -80,69 +113,65 @@ function setMapToDrawLayersOn(map) {
 /**
  * Toggles on displaying an asset on the map.
  *
- * @param {string} layerName
+ * @param {Object} layer Data for layer coming from Firestore
  * @param {google.maps.Map} map main map
  */
-function toggleLayerOn(layerName, map) {
-  const currentLayerMapValue = layerMap.get(layerName);
-  currentLayerMapValue.displayed = true;
-  if (currentLayerMapValue.data) {
-    addLayerFromFeatures(currentLayerMapValue, layerName);
+function toggleLayerOn(layer, map) {
+  const index = layer['index'];
+  const layerMapValue = layerArray[index];
+  layerMapValue.displayed = true;
+  if (layerMapValue.data) {
+    addLayerFromFeatures(layerMapValue, index);
   } else {
-    addLayer(layerName, currentLayerMapValue.index, map);
+    addLayer(layer, map);
   }
 }
 
 /**
- * Toggles off displaying an asset on the map.
- *
- * @param {string} layerName
+ * Toggles off displaying an asset on the map. Sets its displayed attribute
+ * to false, and recreates the layer (for deck) or removes it from overlay array
+ * (for others).
+ * @param {number} index
  * @param {google.maps.Map} map main map
  */
-function toggleLayerOff(layerName, map) {
-  removeLayer(layerName, map);
-}
-
-/**
- * Function object to extract a color from a JSON Feature.
- *
- * @param {GeoJSON.Feature} feature
- * @return {Array} RGBA array
- */
-function getColorOfFeature(feature) {
-  return showColor(feature.properties['color']);
+function toggleLayerOff(index, map) {
+  const layerMapValue = layerArray[index];
+  layerMapValue.displayed = false;
```

Comment:
nit: layerArrayValue ? think there's a couple of these

Reply:
renamed to displayedLayerData, can rename again if you want a different name for the class